app/models/target.rb

```diff
@@ -2,4 +2,13 @@ class Target < ApplicationRecord
   validates :topic, :length, :title, :latitude, :longitude, :user, presence: true
   TOPICS = %w[football travel politics art dating music movies series food].freeze
   belongs_to :user
+
+  def matches
+    radius_query  = "earth_distance(ll_to_earth(#{latitude},#{longitude}),"\
+                    " ll_to_earth(latitude, longitude)) <= (#{length} + length)"
+
+    matching_targets = Target.shared_topic(self).where(radius_query).to_a
```

Comment:
No need to assign `matching_targets` since ruby returns the result of the last line